.github/workflows/update.yaml

```
@@ -0,0 +1,54 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Auto Refresh
+
+on:
+  schedule:
+    - cron: '0 1 * * *'
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Run
+      run: |
+        python src/main.py kcQuests -d 1
+    - name: Copy file
+      run: |
+        cp quests-scn.json dist/quests-scn.json
+        cp quests-scn-new.json dist/quests-scn-new.json
+    # Runs a set of commands using the runners shell
+    - name: Commit changes
+      uses: EndBug/add-and-commit@v9
+      with:
```

I don't think it's a good idea to commit changes when submitting a pull request to the main branch.

```suggestion
      if: github.ref == 'refs/heads/main'
      with:
```

---

https://github.com/kcwikizh/kcQuests/pull/9/commits/a02119e6d77a14bd49fd6d27ba5858ee8d6d0f2d